src/sentry/static/sentry/app/views/onboarding/createProject.jsx

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import {browserHistory, Link} from 'react-router';
+
+import ApiMixin from '../../mixins/apiMixin';
+import ProjectActions from '../../actions/projectActions';
+import {getPlatformName} from './utils';
+
+import OnboardingProject from '../onboarding/project';
+import {t} from '../../locale';
+
+import Raven from 'raven-js';
+
+const CreateProject = React.createClass({
+  propTypes: {
+    getDocsUrl: React.PropTypes.func
+  },
+
+  contextTypes: {
+    organization: React.PropTypes.object,
+    location: React.PropTypes.object
+  },
+
+  mixins: [ApiMixin],
+
+  getDefaultProps() {
+    return {
+      getDocsUrl: ({slug, projectSlug, platform}) =>
+        `/onboarding/${slug}/${projectSlug}/configure/${platform}`
+    };
+  },
+
+  getInitialState() {
+    let {teams} = this.context.organization;
+    let accessTeams = teams.filter(team => team.hasAccess);
+
+    let team =
+      this.context.location.query.team || (accessTeams.length && accessTeams[0].slug);
+    return {
+      loading: true,
+      error: false,
+      platform: '',
+      projectName: '',
+      team: team,
+      inFlight: false
+    };
+  },
+
+  createProject() {
+    let {slug} = this.context.organization;
+    let {projectName, platform, team, inFlight} = this.state;
+
+    //prevent double-trigger
+    if (inFlight) return;
+    this.setState({inFlight: true});
+
+    if (!projectName) {
+      Raven.captureMessage('Onboarding no project name ', {
+        extra: {props: this.props, state: this.state}
+      });
+    }
+
+    this.api.request(`/teams/${slug}/${team}/projects/`, {
+      method: 'POST',
+      data: {
+        name: projectName,
+        platform: platform
+      },
+      success: data => {
+        ProjectActions.createSuccess(data);
+
+        // navigate to new url _now_
+        const url = this.props.getDocsUrl({slug, projectSlug: data.slug, platform});
+        this.setState({inFlight: false});
+        browserHistory.push(url);
+      },
+      error: err => {
+        this.setState({
+          inFlight: false,
+          error: err.responseJSON.detail
+        });
+
+        if (err.status != 403) {
+          Raven.captureMessage('Onboarding project creation failed', {
+            extra: {
+              err,
+              props: this.props,
+              state: this.state
+            }
+          });
+        }
+      }
+    });
+  },
+
+  next() {
+    this.createProject();
+  },
+
+  render() {
+    let {projectName, platform, error} = this.state;
+    let {slug, teams} = this.context.organization;
+    let accessTeams = teams.filter(team => team.hasAccess);
+
+    const stepProps = {
+      next: this.next,
+      platform: platform,
```

Comment:
no change needed here, but thoughts on adding eslint rule for preferring shorthand?

Reply:
initial impression is no but I could be convinced otherwise

Reply:
https://eslint.org/docs/rules/object-shorthand
* can be `--fix`ed
* cleaner

Reply:
👍  I've given people this feedback repeatedly, if we can automate it, let's.